src/software/ai/hl/stp/tactic/defense_shadow_enemy_tactic.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "software/ai/hl/stp/evaluation/enemy_threat.h"
+#include "software/ai/hl/stp/tactic/tactic.h"
+
+/**
+ * The ShadowEnemyTactic will shadow and mark the robot specified in the given
+ * EnemyThreat. It will choose to either block the enemy's shot on net or the pass it
+ * would receive from another enemy.
+ */
+class DefenseShadowEnemyTactic : public Tactic
+{
+   public:
+    /**
+     * Creates a new DefenseShadowEnemyTactic
+     *
+     * @param field The field
+     * @param friendly_team The friendly team
+     * @param enemy_team The enemy team
+     * @param ball The ball
+     * @param ignore_goalie Whether or not to ignore the friendly goalie when determining
+     * the best position to shadow in order to block shots
+     * @param shadow_distance How far away to shadow the enemy from
+     * @param enemy_team_can_pass Whether or not the enemy team can pass
+     * @param loop_forever Whether or not this tactic should loop forever
+     */
+    explicit DefenseShadowEnemyTactic(const Field &field, const Team &friendly_team,
+                                      const Team &enemy_team, const Ball &ball,
+                                      bool ignore_goalie, double shadow_distance,
+                                      bool enemy_team_can_pass, bool loop_forever = true);
+
+    std::string getName() const override;
+
+    /**
+     * Updates the world parameters for this Tactic.
+     *
+     * @param field The current state of the field
+     * @param friendly_team The current state of the friendly team
+     * @param enemy_team The current state of the enemy team
+     * @param ball The current state of the ball
+     */
+    void updateWorldParams(const Field &field, const Team &friendly_team,
+                           const Team &enemy_team, const Ball &ball);
+
+    /**
+     * Updates the control parameters for this Tactic
+     *
+     * @param enemy_threat The EnemyThread this Tactic is shadowing
+     */
+    void updateControlParams(const Evaluation::EnemyThreat &enemy_threat);
+
+    double calculateRobotCost(const Robot &robot, const World &world) override;
```

Comment:
Nit: Can we add javadocs comment here

Reply:
I generally don't add javadocs to functions marked `override` since the main comment is in the superclass / interface defining the function. The purpose of the function doesn't change to I prefer not to make redundant comments. This is the case for most of the codebase. Do you agree with that?

Reply:
I agree with Mat on this.

Reply:
I see what you mean. I've been adding comments more specific to the function override

Reply:
Hmm, could you provide some examples? A javadoc comment should only define the interface, not explain the implementation, and since an override is supposed to have the same interface, I'm struggling to imagine why we'd want a javadoc for it.

Reply:
It could provide additional context for how it behaves. For example I javadoc straightline path planner to say what sort of behaviour we expect.

Reply:
I would generally classify those things as implementation-specific. Eg. The path planner interface is that is returns "paths". The straightline planner also returns paths, but it's an implementation detail that these paths are straight lines (or splines or whatever). Those comments should probably live either inside the function or as a class comment if it's a class. We can come back to that stuff later.